### PR TITLE
feat: token deposit

### PR DESCRIPTION
**Pull request by obiyankenobi**
image
image
image
image
image
image

### CODE REVIEW

**Comment by pedroferreira1** [comment]
Still missing error screenshot

File: src/screens/CreateTokenDepositNotice.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            <Text style={this.style.text}>
+              When creating new tokens, a{' '}
+              <Text style={this.style.link}>
+                deposit of {hathorLib.tokens.getDepositPercentage() * 100}%{' '}
+              </Text>
+              in HTR is required - e.g. if you create 1000 NewCoins, 10 HTR are needed as deposit.
+            </Text>
+            <Text style={this.style.text}>
+              If these tokens are later melted, the HTR deposit will be returned. Read more about it{' '}
+              <Link href='https://gitlab.com/HathorNetwork/rfcs/blob/master/text/0011-token-deposit.md'>here</Link>
```
Comment (msbrogli) [inline src/screens/CreateTokenDepositNotice.js]: Move the URL to the config file.

File: src/screens/CreateTokenName.js [hunk 2]
```diff
@@ -46,11 +48,19 @@ class CreateTokenName extends React.Component {
...
-            <NewHathorButton
-              title='Next'
-              disabled={!this.state.name}
-              onPress={this.onButtonPress}
-            />
+            <View>
+              <InfoBox
+                items={[
+                  <Text>Token name should be the full name of the new token you are creating</Text>,
+                  <Italic>E.g. Hathor</Italic>
```
Comment (msbrogli) [inline src/screens/CreateTokenName.js]: I would use another example. Something with two words and a space. Maybe "My Custom Token" or "Hathor Network".

File: src/screens/CreateTokenSymbol.js [hunk 3]
```diff
@@ -38,28 +39,49 @@ class CreateTokenSymbol extends React.Component {
...
   }
 
+  isButtonDisabled = () => {
+    if (!this.state.symbol) {
+      return true;
+    }
+
+    if (this.state.symbol.length < 2) {
+      return true;
+    }
```
Comment (msbrogli) [inline src/screens/CreateTokenSymbol.js]: What about the max length of 5?

File: src/screens/CreateTokenName.js (hunk 2 above)
Comment (obiyankenobi) [reply src/screens/CreateTokenName.js]: The problem is that you're leading the person to use that model. I prefer without spaces: 'YanCoin' or 'MyCustomToken'. I used a single word to make sure we don't prime the user with a preference.

We can also give 2 examples, with and without space.

File: src/screens/CreateTokenSymbol.js (hunk 3 above)
Comment (obiyankenobi) [reply src/screens/CreateTokenSymbol.js]: We already forbid that in the input field, bu I'll change to also enforce here.

File: src/screens/CreateTokenName.js (hunk 2 above)
Comment (msbrogli) [reply src/screens/CreateTokenName.js]: Seems good. Examples: "MyCustomToken", "My Custom Token", etc.

File: src/screens/CreateTokenName.js [hunk 4]
```diff
@@ -46,11 +48,19 @@ class CreateTokenName extends React.Component {
               onChangeText={this.onNameChange}
               value={this.state.name}
```
Comment (pedroferreira1) [inline src/screens/CreateTokenName.js]: Should we validate that this input has at most 30 chars? Just to keep it consistent with the full node validation

Comment (obiyankenobi) [reply src/screens/CreateTokenName.js]: That's true, we're not enforcing it here. Will do it.

Comment (obiyankenobi) [reply src/screens/CreateTokenName.js]: Did that and added a 'counter' for the name:
image

File: src/screens/CreateTokenSymbol.js (hunk 3 above)
Comment (obiyankenobi) [reply src/screens/CreateTokenSymbol.js]: Done

File: src/screens/CreateTokenName.js (hunk 2 above)
Comment (obiyankenobi) [reply src/screens/CreateTokenName.js]: Done

File: src/screens/CreateTokenDepositNotice.js (hunk 1 above)
Comment (obiyankenobi) [reply src/screens/CreateTokenDepositNotice.js]: Done

**Pull request merged**